src/language_server_protocol.rs

```diff
@@ -2018,6 +2019,32 @@ impl LanguageClient {
         Ok(())
     }
 
+    pub fn workspace_configuration(&self, params: &Value) -> Result<Value> {
+        let config_params = ConfigurationParams::deserialize(params)?;
+
+        let filename = self.vim()?.get_filename(&Value::Null)?;
+        let language_id = self.vim()?.get_language_id(&filename, &Value::Null)?;
+        let root = self.get(|state| state.roots.get(&language_id).cloned().unwrap_or_default())?;
+
+        let settings = self.get_workspace_settings(&root)?;
+
+        let configuration_items = config_params
+            .items
+            .into_iter()
+            .map(|item| {
+                let mut settings_val = &settings;
+
+                for item in item.section.unwrap().split('.') {
```

Comment:
The `unwrap` here is a little dangerous. I have a few suggestions around this. 

- I'd call `.filter(|item| item.section.is_some())` before the call to `.map()` to have that `unwrap` be safe.
- The unwinding of the section you are doing there can probably be achieve more easily by using `settings.pointer`, I would change the body of that map to something like this:
    ```
    let section = format!("/{}", item.section.unwrap().replace(".", "/"));
    settings
        .pointer(section.as_str())
        .cloned()
        .unwrap_or(Value::Null)
    ```
   Note that, as per the specification, null must be returned if the section is not found in the config.
- this will also no longer return a Result on the map, so no need to collect into `Result<Value>` anymore.